### DISCUSSION TITLE
.env里声明的环境变量怎么注入到业务代码里

### DISCUSSION THREAD

**Discussion by PPPeony** [NervJS/taro#16754] [Q & A] [answered]
.env里声明的环境变量怎么注入到业务代码里？
我在.env.dev里声明了一个环境变量
`env
TARO_APP_CONFIG_WINDOW_NAV_BAR_TITLE="智能助理"
`
然后我在src/app.config.ts中引入了这个环境变量
`js
 navigationBarTitleText: process.env.TARO_APP_CONFIG_WINDOW_NAV_BAR_TITLE,
`
为什么打包出来app.json没有这个环境变量的值呀

**Comment by TheKonka** [comment]
https://docs.taro.zone/en/docs/env-mode-config/#%E9%BB%98%E8%AE%A4%E8%A1%8C%E4%B8%BA
```
.env.development     # 在 development 模式时被载入
.env.production      # 在 production 模式时被载入
```
不是 .env.dev

**Reply by PPPeony** [reply, marked as answer]
解决问题了，我参考的是这个帖子https://github.com/NervJS/taro/discussions/14330
我将taro的版本号升级到3.6+，自定义模式的环境变量也可以注入了。

**Comment by PPPeony** [comment]
文档上也写了3.5.10以上版本的taro才支持环境变量注入，但是我没注意看，然后项目里的版本号是精确版本号，所以怎么配环境变量也是注入不到代码里的😭

**Reply by TheKonka** [reply]
3.5.10那个是通过 .env 文件注入的，如果是通过编译配置文件里面的`env`或者`defineConstants`注入是一直都可以的
